train_gpt2.cu

```diff
@@ -686,45 +690,61 @@ void gpt2_forward(GPT2 *model, const int* inputs, const int* targets, size_t B,
     }
 
     matmul_forward_cublaslt(acts.output, acts.lnf, params.wte, NULL, B, T, C, Vp, main_stream);
+    cudaCheck(cudaDeviceSynchronize());
```

Comment:
curious why

Reply:
it might not be necessary, but  I think for now, a programming model where every gpt2_... function guarantees that all its work is done by the time the function returns hopefully reduces the chances for errors.